packages/opentelemetry-exporter/src/index.js

```diff
@@ -0,0 +1,288 @@
+  broker. Unlike client and server, there is no direct critical path latency
+  relationship between producer and consumer spans.
+  PRODUCER = 3,
+
+  Indicates that the span describes consumer receiving a message from a
+  broker. Unlike client and server, there is no direct critical path latency
+  relationship between producer and consumer spans.
+  CONSUMER = 4
+*/
+const SPAN_KIND_INTERNAL = 0;
+const SPAN_KIND_SERVER = 1;
+const SPAN_KIND_CLIENT = 2;
+const SPAN_KIND_PRODUCER = 3;
+const SPAN_KIND_CONSUMER = 4;
+const SERVER_KIND = 'server';
+const CLIENT_KIND = 'client';
+const PRODUCER_KIND = 'producer';
+const CONSUMER_KIND = 'consumer';
+const INTERNAL_KIND = 'internal';
+/**
+ * https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto#L314
+ * Status Code
+ *
+ * Default
+ * UNSET = 0,
+ * The operation has been validated by an Application developer or
+ * Operator to have completed successfully.
+ * OK = 1,
+ * The operation contains an error.
+ * ERROR = 2
+ */
+
+const DEPRECATED_STATUS_CODE_OK = 0;
+const STATUS_UNSET = 0;
+const STATUS_OK = 1;
+// const STATUS_ERROR = 2;
+
+class InstanaExporter {
+  /**
+   * @param {{ agentKey?: string, endpointUrl?: string }}
+   */
+  constructor({ agentKey, endpointUrl }) {
+    this._isShutdown = false;
+    // If endpoint URL and agent key are not provided, we don't set them in an attempt to use env vars, if they were set
+    if (endpointUrl) {
+      process.env[instanaEndpointUrlEnvVar] = endpointUrl;
+    }
+    if (agentKey) {
+      process.env[instanaAgentKeyEnvVar] = agentKey;
+    }
+
+    if (!process.env[instanaEndpointUrlEnvVar] || !process.env[instanaAgentKeyEnvVar]) {
+      throw new Error('The Instana endpoint URL and/or the agent key were not provided');
+    }
+
+    instanaEnvironment.validate();
+    instanaBackendConnector.init();
+  }
+
+  /**
+   * @param {Array.<ReadableSpan>} spans
+   * @param {(result: import('@opentelemetry/core').ExportResult) => void} resultCallback
+   */
+  export(spans, resultCallback) {
+    const validSpans = spans.map(this._transform.bind(this));
+
+    if (!this._isShutdown) {
+      this._sendSpans(validSpans, resultCallback);
+    }
+  }
+
+  /**
+   * @returns {Promise.<void>}
+   */
+  shutdown() {
+    /**
+     * In the future, there may be the need to "end" any kind of process or network communications with our backend.
+     * Right now, there isn't seem to be any situation where we need that, as we simply make HTTP requests without
+     * actually making any type of stateful or persistent connections.
+     */
+    this._isShutdown = true;
+    return Promise.resolve();
+  }
+
+  /**
+   * @param {Array.<InstanaBaseSpan>} spans
+   * @param {(result: import('@opentelemetry/core').ExportResult) => void} resultCallback
+   */
+  _sendSpans(spans, resultCallback) {
+    instanaBackendConnector.sendSpans(spans, (err /* , _responseBody */) => {
```

Comment:
The backend connector's init method would need to be called with `propagateErrorsUpstream = true` otherwise the `sendSpans` method will not pass errors to the given callback. (default for propagateErrorsUpstream is false)

Reply:
But on the other hand: Do we even want to report an error to OTel's resultCallback? What would be the consequence.

Reply:
I will add this test. But from my experience so far, the exporter always fails silently to prevent the original code to crash. This is why I had to add temporary try/catch statements when I started the creation of the exporter.
I don't quite understand what the error reported via `resultCallback` does, but definitely does not crash the application.